packages/govuk-frontend/src/govuk/common/index.mjs

```diff
@@ -193,10 +220,14 @@ export function setFocus($element, options = {}) {
  *
  * @internal
  * @param {Element} $module - HTML element to be checked
+ * @param {string} moduleName - name of component module
  * @returns {boolean} Whether component is already initialised
  */
-export function isInitialised($module) {
-  return $module instanceof HTMLElement && 'moduleInit' in $module.dataset
+export function isInitialised($module, moduleName) {
+  return (
+    $module instanceof HTMLElement &&
+    `${kebabCaseToCamelCase(moduleName)}Init` in $module.dataset
```

Comment:
```suggestion
$module.hasAttribute(`data-${moduleName}-init`)
```

We can avoid the `kebabCaseToCamelCase` if we look for the attribute. Also wondering if that function wouldn't be better exported by same module as the base component class one, to regroup the concept for computing the name of that attribute in a single file.